### DISCUSSION TITLE
db.all() shows database as empty when it's not

### DISCUSSION THREAD

**Discussion by MrPigss** [msiemens/tinydb#457] [Q&A] [answered]
Assume you have a db with multiple tables each with multiple documents.
db:
- Table: a
- Table: b
- Table: c

You now open an instance of TinyDb with that db:

```Python

with TinyDB("databasefile.db") as db:
    db.all() #returns []
```

Because TinyDB passes the call to the TinyDB instance directly to the default Table (that might or might not be used) it returns an empty array.
this makes it seem as if the database itself is empty.
While passing the calls to the default db is intentional design i find it confusing at times where you don't work with a single default db.

Under the moto:
> Explicit is better than implicit

Why don't we have to explicitly select the table we are working with?

LMKWYT

**Comment by msiemens** [comment, marked as answer]
> Why don't we have to explicitly select the table we are working with?

Simply put, it's for historical reasons 🙂 The very first design of TinyDB had only a single table. Relatively quickly I found that I needed to store multiple tables within the same database so I added the possibility to create new tables. But, crucially, in my mind (back in 2013) I thought that most use cases would probably only use a single table and having to explicitly create a table when one doesn't need multiple tables felt like an unnecessary burden to me.

Now, almost 10 years later I agree that explicitly creating tables would indeed save a lot of confusion when people who used to use a single table start using multiple tables. But I really can't count how many projects need a single vs. multiple tables and at this point I fear that changing this would probably cause more confusion/code breakage than it would prevent.

**Reply by msiemens** [reply]
Quick note: If you're curious about the very early design of TinyDB prior to its 1.0 release, you can head over to https://github.com/msiemens/TinyDB-Archive where I've uploaded the Git history of TinyDB prior to 1.0. Fun fact: the very first iteration of TinyDB was called PlainDB 🙂 

The commit that introduced tables to TinyDB a.k.a. PlainDB: https://github.com/msiemens/TinyDB-Archive/commit/b98f37c4f55349a155d44d000c3b019796d07478
